src/rdata/naptr.rs

```diff
@@ -0,0 +1,545 @@
+//! Record data from [RFC 3403]: NAPTR records.
+//!
+//! This RFC defines the Naptr record type.
+//!
+//! [RFC 3403]: https://tools.ietf.org/html/rfc3403
+
+use crate::base::{
+    name::FlattenInto,
+    rdata::ComposeRecordData,
+    scan::{Scan, Scanner},
+    wire::{Compose, Parse, ParseError},
+    zonefile_fmt::{self, Formatter, ZonefileFmt},
+    CanonicalOrd, CharStr, ParseRecordData, ParsedName, RecordData, Rtype,
+    ToName,
+};
+use core::{cmp::Ordering, fmt, hash};
+#[cfg(feature = "serde")]
+use octseq::builder::{EmptyBuilder, FromBuilder, OctetsBuilder};
+use octseq::{Octets, OctetsFrom, OctetsInto, Parser};
+
+//------------ Naptr ---------------------------------------------------------
+
+#[derive(Clone)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(bound(
+        serialize = "
+            Octs: octseq::serde::SerializeOctets + AsRef<[u8]>,
+            Name: serde::Serialize,
+        ",
+        deserialize = "
+            Octs: FromBuilder + octseq::serde::DeserializeOctets<'de>,
+            <Octs as FromBuilder>::Builder:
+                OctetsBuilder + EmptyBuilder
+                + AsRef<[u8]> + AsMut<[u8]>,
+            Name: serde::Deserialize<'de>,
+        ",
+    ))
+)]
+pub struct Naptr<Octs, Name> {
+    order: u16,
+    preference: u16,
+    flags: CharStr<Octs>,
+    services: CharStr<Octs>,
+    regexp: CharStr<Octs>,
+    replacement: Name,
+}
+
+impl Naptr<(), ()> {
+    /// The rtype of this record data type.
+    pub(crate) const RTYPE: Rtype = Rtype::NAPTR;
+}
+
+impl<Octs, Name> Naptr<Octs, Name> {
+    pub fn new(
+        order: u16,
+        preference: u16,
+        flags: CharStr<Octs>,
+        services: CharStr<Octs>,
+        regexp: CharStr<Octs>,
+        replacement: Name,
+    ) -> Self {
+        Naptr {
+            order,
+            preference,
+            flags,
+            services,
+            regexp,
+            replacement,
+        }
+    }
+
+    pub fn order(&self) -> u16 {
```

Comment:
Please add RustDoc comments to each of these public functions, e.g. see `src/rdata/rfc1035/soa.rs`.

Reply:
So, the SOA documentation uses the URL from https://tools.ietf.org/html/rfc1035#section-3.3.13 in the file located at https://github.com/NLnetLabs/domain/blob/984d16b6d8f6eb925124b927d58a8c5ddf50d08f/src/rdata/rfc1035/soa.rs#L30. Should I use the https://tools.ietf.org/html/rfc3403 or https://www.rfc-editor.org/rfc/rfc3403.html in doc?

Reply:
We prefer the https://www.rfc-editor.org/rfc/rfc3403 format.